autogpt_platform/backend/backend/data/onboarding.py

```diff
@@ -95,43 +94,42 @@ async def update_user_onboarding(user_id: str, data: UserOnboardingUpdate):
 
 
 async def reward_user(user_id: str, step: OnboardingStep):
-    async with db.locked_transaction(f"usr_trx_{user_id}-reward"):
```

Comment:
@kcze why do we need to lock here? where `onboarding_reward` is already atomic with a key